[Doc] Update Mapbox docs regarding deck v9 and react-map-gl v7 (#8146)
visgl/deck.gl: docs/api-reference/mapbox/overview.md

# @deck.gl/mapbox

> This module makes it easy to use deck.gl as native layers and controls in the Mapbox and Maplibre GL JS ecosystem. 

Reviewer: The `native` part seems a bit confusing?
```suggestion
This module makes integrates deck.gl layers and controls into the Mapbox and Maplibre GL JS plugin ecosystem. 
```

> > For brevity, the term 'mapbox-gl' will refer collectively to any library that is compatible with Mapbox GL JS, including but not limited to maplibre-gl. Known exceptions will be clearly marked. For more, see [Compatibility with Mapbox GL JS forks](../../developer-guide/base-maps/using-with-mapbox.md#compatibility-with-mapbox-gl-js-forks)

Reviewer: I would probably keep typing out `mapbox-API compatible library` in the text rather than focus on brevity.

## Use Cases

> One should note that this module is *not required* to use mapbox-gl as a base map for deck.gl. When you use this module, Mapbox is the root element and deck.gl is the child, with Mapbox handling all user inputs. Some of deck.gl's features are therefore unavailable due to limitations of mapbox-gl's API, see [limitations](#limitations) and [compatibility](#compatibility) below. If you just want the base map as a back drop, and do not need mapbox-gl's UI controls or mixing deck and Mapbox layers, it is recommended that you use deck.gl as the root element. Visit the [mapbox base map developer guide](../../developer-guide/base-maps/using-with-mapbox.md) for examples of each option.

Reviewer: I think the root element discussion should be separate sentience after the sentence explaining of the usage options. Jumping into root element directly seems overly technical. I would probably also write out `HTML element` to make it a little bit less cryptic.

It may be easier to understand the concepts of the module if you are already a mapbox-gl developer.

### Using mapbox-gl controls with deck.gl

> The Mapbox ecosystem offers many well-designed controls, from the basic functionalities of `NavigationControl`, `Popup` and `GeolocateControl`, to vendor-service-bound UI implementations such as `mapbox-gl-geocoder` and `mapbox-gl-directions`. These libraries require that the Mapbox Map holds the source of truth of the camera state, instead of the normal [state management](../../developer-guide/interactivity.md) by `Deck`. When you use the `MapboxOverlay` or `MapboxLayer` classes from this module, deck.gl plays nice with all the mapbox-gl peripherals.

Reviewer: I think having a section about the source of truth of the camera system first would be useful as this is the two major modes. That's where the details about "root" elements could go.

### Mixing deck.gl layers and Mapbox layers

> One major use case for interleaving deck.gl and Mapbox is that some important information in the Mapbox map could be hidden by a deck.gl visualization layer, and controlling opacity is not enough. A typical example of this is labels and roads, where it is desirable to have a deck.gl visualization layer render on top of the Mapbox geography, but where one might still want to see e.g. labels and/or roads. Alternatively, the deck.gl visualization should cover the ground, but not the roads and labels.

Reviewer: Drop the "one major use case". Too much verbiage to get to the point.

To inject a deck layer into the Mapbox stack, either:

- Add a `beforeId` prop to any layer passed to the [MapboxOverlay](./mapbox-overlay.md) control (recommended).
- Create a [MapboxLayer](./mapbox-layer.md) and call the [`map.addLayer(layer, before?)`](https://www.mapbox.com/mapbox-gl-js/api/#map#addlayer) API.

Mapbox provides an example of [finding the first label layer](https://www.mapbox.com/mapbox-gl-js/example/geojson-layer-in-stack/). For more sophisticated injection point lookups, refer to Mapbox' documentation on the format of Mapbox style layers, see [Mapbox Style Spec](https://www.mapbox.com/mapbox-gl-js/style-spec/#layers).

In some cases, the application wants to add a deck.gl 3D layer (e.g. ArcLayer, HexagonLayer, GeoJsonLayer) on top of a Mapbox basemap, while seamlessly blend into the z-buffer. This will interleave the useful visualization layers from both the deck.gl and Mapbox layer catalogs. In this case, a `beforeId` is not needed.

> ## Compatibility

Reviewer 2: I looked for the first release the WebGL2 compatibility commit appeared to make this table:
- Mapbox `useWebGl` flag: https://github.com/mapbox/mapbox-gl-js/commit/d7652ba8eda70a97809840364beed81bad07930d
- Mapbox by default: https://github.com/mapbox/mapbox-gl-js/commit/27aa25082af00c83c25e92bff87c001c0137aafd
- Maplibre by default: https://github.com/maplibre/maplibre-gl-js/commit/7a1d5b5355f9b1da31ced9d6f1731b9828f8e057

Libraries utilizing WebGL1 cannot be interleaved with deck.gl. deck.gl has phased out support for WebGL1, requiring WebGL2 at a minimum. The following table details WebGL support across different versions of mapbox-gl and maplibre-gl.

| Library                       | WebGL Version   | Overlay Support | Interleaved Support |
|-------------------------------|-----------------|-----------------|---------------------|

> | mapbox-gl (before v2.13)      | WebGL1          | Yes             | No                  |

Reviewer: Suggest we put the compatible libraries at the top of the list. 

Nit: For a bit of fun and color, I use ✅ ❌ unicode instead of yes no in e.g. luma and loaders docs.
